src/attachmentlistmodel.cpp

```diff
@@ -148,15 +148,12 @@ void AttachmentListModel::onAttachmentDownloadProgressChanged(const QString &att
 
 void AttachmentListModel::onAttachmentPathChanged(const QString &attachmentLocation, const QString &path)
 {
+    // path is coming from the agent and guaranteed to be a valid existing file.
     for (int i = 0; i < m_attachmentsList.count(); ++i) {
         Attachment *attachment = m_attachmentsList.at(i);
         if (attachment->location == attachmentLocation) {
-            // Make url more secure and compatible with QMF client
-            QString downloads = downloadFolder(m_message, attachmentLocation);
-            QString fileName = path;
-            QString secureUrl = QUrl::fromLocalFile(downloads + "/" + fileName.remove(downloads).remove('/')).toString();
-            if (attachment->url != secureUrl) {
-                attachment->url = secureUrl;
+            if (attachment->url != path) {
+                attachment->url = path;
```

Comment:
So effectively reverting part of commit bddff0f196d4 here. 

The commit message on that was somewhat short but the problem was that the email could claim some "../../foo" type of filename in the content and the the code in this class was reading the file elsewhere. 

But indeed looks like the path here is coming from either EmailAgent after saving something or from the QFileSystemWatcher here (as in ::resetMode()) which should(?) filter the result with attachmentPath().  cc @mkenttala and @llewelld